spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -182,13 +183,28 @@ private Schema schemaWithMetadataColumns() {
   public Scan build() {
     Long snapshotId = readConf.snapshotId();
     Long asOfTimestamp = readConf.asOfTimestamp();
+    String branch = readConf.branch();
+    String tag = readConf.branch();
+
+    Preconditions.checkArgument(
+        branch == null || tag == null,
+        "Cannot set both %s and %s to select which table snapshot to scan",
+        SparkReadOptions.BRANCH,
+        SparkReadOptions.TAG);
 
     Preconditions.checkArgument(
         snapshotId == null || asOfTimestamp == null,
         "Cannot set both %s and %s to select which table snapshot to scan",
         SparkReadOptions.SNAPSHOT_ID,
         SparkReadOptions.AS_OF_TIMESTAMP);
 
+    String snapshotRef = branch != null ? branch : tag;
+    Preconditions.checkArgument(
```

Comment:
I don't think that we need checks here. The `TableScan` will check for consistency and throw exceptions, so we should allow it to fail there rather than duplicating checks. We can probably remove both this and the one checking for both tag and branch set. We still need tests for this behavior though.